File: 02_Object_Oriented_Programming/02_Class_Relationships/03_Aggregation/python/main.py
"""
Driver — demonstrates Aggregation (HAS-A with weak ownership).
Run: python main.py
"""

from player import Player
from team import Team


def main() -> None:
    p1 = Player("Stephen", 30)
    p2 = Player("Klay", 11)
    p3 = Player("Draymond", 23)

    team = Team("Warriors")
    team.add_player(p1)
    team.add_player(p2)
    team.add_player(p3)
    team.show_team()

    # Remove a player — the player object still exists
    team.remove_player(p2)
    print("\nAfter removing Klay:")
    team.show_team()
    print(f"\nKlay still exists: {p2}")


if __name__ == "__main__":
    main()

"""
Expected Output:
Team Warriors has players:
  - Stephen (#30)
  - Klay (#11)
  - Draymond (#23)

After removing Klay:
Team Warriors has players:
  - Stephen (#30)
  - Draymond (#23)

Klay still exists: Klay (#11)
"""

File: 02_Object_Oriented_Programming/02_Class_Relationships/03_Aggregation/javascript/main.js
/**
 * Driver — demonstrates Aggregation (HAS-A with weak ownership).
 * Run: node main.js
 */

const Player = require("./player");
const Team = require("./team");

const p1 = new Player("Stephen", 30);
const p2 = new Player("Klay", 11);
const p3 = new Player("Draymond", 23);

const team = new Team("Warriors");
team.addPlayer(p1);
team.addPlayer(p2);
team.addPlayer(p3);
team.showTeam();

// Remove a player — the player object still exists
team.removePlayer(p2);
console.log("\nAfter removing Klay:");
team.showTeam();
console.log(`\nKlay still exists: ${p2}`);

/*
Expected Output:
Team Warriors has players:
  - Stephen (#30)
  - Klay (#11)
  - Draymond (#23)

After removing Klay:
Team Warriors has players:
  - Stephen (#30)
  - Draymond (#23)

Klay still exists: Klay (#11)
*/
